parcels/particle.py

```diff
@@ -15,25 +15,55 @@
 def AdvectionRK4(particle, grid, time, dt):
     f_lat = dt / 1000. / 1.852 / 60.
     f_lon = f_lat / math.cos(particle.lat*math.pi/180)
-    u1 = grid.U[time, particle.lon, particle.lat]
-    v1 = grid.V[time, particle.lon, particle.lat]
-    lon1, lat1 = (particle.lon + u1*.5*f_lon, particle.lat + v1*.5*f_lat)
-    u2, v2 = (grid.U[time + .5 * dt, lon1, lat1], grid.V[time + .5 * dt, lon1, lat1])
-    lon2, lat2 = (particle.lon + u2*.5*f_lon, particle.lat + v2*.5*f_lat)
-    u3, v3 = (grid.U[time + .5 * dt, lon2, lat2], grid.V[time + .5 * dt, lon2, lat2])
-    lon3, lat3 = (particle.lon + u3*f_lon, particle.lat + v3*f_lat)
-    u4, v4 = (grid.U[time + dt, lon3, lat3], grid.V[time + dt, lon3, lat3])
+    posvertdir = -1.  # TODO: Decide on direction of positive w
+    u1 = grid.U[time, particle.lon, particle.lat, particle.dep]
+    lon1 = particle.lon + u1*.5*f_lon
+    v1 = grid.V[time, particle.lon, particle.lat, particle.dep]
+    lat1 = particle.lat + v1*.5*f_lat
+    if hasattr(grid, 'W'):
```

Comment:
I really don't like the extreme special casing in this routine; it goes against the entire philosophy of writing simple, easy-to-understand kernels. And also, `hasattr` is a pure Python construct that has no real equivalent in C code. 

However, since we don't have a dimension-independent "position" abstraction yet (ie. a vector variable, instead of several scalar variables), I would argue we simply always compute `w` and let it compute a whole lot of zeros for now. We then really need to create a "vector variable" abstraction though before we can don any sincere benchmarking, since we want to get rid of this blatant performance overhead.